.circleci/config.yml

```diff
@@ -85,12 +85,19 @@ workflows:
                 - 3.1.6
                 - 3.2.6
                 - 3.3.6
+                - 3.4.7
               gemfile:
                 - gemfiles/6.1.gemfile
                 - gemfiles/7.0.gemfile
                 - gemfiles/7.1.gemfile
                 - gemfiles/7.2.gemfile
                 - gemfiles/8.0.gemfile
+                - gemfiles/8.1.gemfile
             exclude:
+              # Rails 8.x requires Ruby >= 3.3
```

Comment:
Since Rails 6.1, 7.0, 7.1, and Ruby 3.1 are EOL, we can remove some of these exclusions. I think that would also address the failing tests. Would you mind removing support for those EOL versions?